### PR TITLE
Add table of contents for src/dmd

### PR DESCRIPTION
**Pull request by dkorpel**
As suggested by Walter: https://github.com/dlang/dmd/pull/9844#issuecomment-495951635

Replaces the old `readme.txt` with a `README.md` that includes a table of contents which should help people navigating the source code.

### CODE REVIEW

**Review by Geod24 — APPROVED** [review]
Very nice! Thanks for doing this. Let's wait a day or two to see if anyone else chime in, especially Walter, but IMO this is good to go.

File: src/dmd/README.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+This is the source code to the DMD compiler
+for the D Programming Language defined in the documents at
+http://dlang.org/
+
+These sources are free, they are redistributable and modifiable
+under the terms of the Boost Software License, Version 1.0.
+The terms of this license are in the file boostlicense.txt,
+or see http://www.boost.org/LICENSE_1_0.txt.
+
+If a particular file has a different license in it, that overrides
```
Comment (PetarKirov) [inline src/dmd/README.md]: Can you check if that's the case for any file? If not it would be better to remove the paragraph.

Comment (Geod24) [reply src/dmd/README.md]: backend/dcomplex is public domain
backend/md5 is licensed to RSA

**Comment by dkorpel** [comment]
It's been some days, can this be merged?

**Comment by Geod24** [comment]
Hey looks like you fixed https://issues.dlang.org/show_bug.cgi?id=14333 in the process, thanks!

**Comment by dkorpel** [comment]
I don't think that refers to the same README, looking at `dmd2/windows/bin/README.txt` on my Windows installation reveals:
```
Win32 files
-----------
dmd.exe		D compiler			http://dlang.org/dmd-windows.html
sc.ini		D compiler configuration file	http://dlang.org/dmd-windows.html#sc_ini
ddemangle.exe	D symbol demangler
dub.exe		D's package manager		https://github.com/dlang/dub
dustmite.exe	D source code minimizer		https://github.com/CyberShadow/DustMite
lib.exe		Librarian			http://www.digitalmars.com/ctg/lib.html
libcurl.dll	Curl Library			https://curl.haxx.se/
lld-link.exe	LLVM Linker			https://lld.llvm.org/
make.exe	Make program			http://www.digitalmars.com/ctg/make.html
optlink.exe	Linker				http://www.digitalmars.com/ctg/optlink.html
rdmd.exe	Run D program as a script	http://dlang.org/rdmd.html
replace.exe	Find/replace text in file(s)
shell.exe	Test script runner		http://www.digitalmars.com/ctg/shell.html
```

In any case it does not look woefully out of date.

**Pull request merged**